chartfx-chart/src/main/java/de/gsi/chart/XYChart.java

```diff
@@ -257,8 +257,11 @@ public void updateAxisRange() {
         // lock datasets to prevent writes while updating the axes
         ObservableList<DataSet> dataSets = this.getAllDatasets();
         // check that all registered data sets have proper ranges defined
-        dataSets.parallelStream().forEach(dataset -> dataset.getAxisDescriptions().parallelStream().filter(axisD -> !axisD.isDefined()) //
-                                                             .forEach(axisDescription -> dataset.lock().writeLockGuard(() -> dataset.recomputeLimits(axisDescription.getDimIndex()))));
+        dataSets.parallelStream()
+                .filter(DataSet::isVisible) // TODO (ennerf): this does not get the dataset out of the axis. Is there a general way to do this w/o modifying the dataset itself ?
```

Comment:
I think a good place to add the `.filter(DataSet::isVisible)` statement would be in `XYChart`
```
protected static void updateNumericAxis(final Axis axis, final List<DataSet> dataSets) {/*...*/}
```
This is the place where the chart loops through all DataSets that are relevant for a given axis and updates the auto-range and auto-grow-ranging.